### ISSUE TITLE
[Bug]: `export * as Foo from "./foo"` doesn't work when containing file and `./foo/index.ts` are entry points

### ISSUE THREAD

**Issue by hwallis93** [Dschungelabenteuer/vite-plugin-entry-shaking#53]
### What does the bug relate to?

- [X] Plugin's core
- [ ] Debugger

### Describe the bug

Minimal repro here - https://github.com/hwallis93/plugin-entry-shaking-bug-repro/

Say I've got a repo where a barrel file does `export * as Hello from "./hello"` where `./hello/index.ts` is another barrel file, where both barrel files are shaken by EntryShakingPlugin, then `Hello` is removed entirely at build time, even if it's used. This causes errors when trying to call functions it contains

E.g. for this directory structure:

```
.
├── index.html
├── package.json
├── src
│   ├── lib
│   │   ├── hello
│   │   │   ├── hello.ts
│   │   │   └── index.ts
│   │   └── index.ts
│   └── main.ts
├── tsconfig.json
└── vite.config.mjs
```

`lib/index.ts` contains:

```typescript
export * as Hello from "./hello"
```

and `lib/hello/index.ts` contains

```typescript
export {helloWorld} from "./hello"
```

then `main.ts` can't consume `Hello`:

```typescript
import {Hello} from "./lib"

// This errors because Hello is not defined
Hello.helloWorld() 
```

### To Reproduce

- Clone minimal repro
- Run `npm install` then `npx vite`
- Note in the console that `Hello.helloWorld is not a function`
- Comment out the EntryShakingPlugin in `vite.config.mjs`
- Note that the error is gone and the code works

### Vite version

5.4.1

**Comment by Dschungelabenteuer** [comment]
Hey! Thank you for that straight-forward repro, it was really helpful. Cross-entries should be supported but it turns out there's an issue when dealing with wildcards across different entries.  Right now, whenever this plugin encounters a wildcard import from another entry, it returns a reformatted `import * from '{path/to/other/entry}'`; it doesn't take that other entry's analysis into account, meaning it tries to import entities from cleaned up entries. Such cleaned-up entries may not expose these entities anymore, resulting in a unresolvable entity. That's your scenario. 

> I import an entity from an entry file which wildcard-imports it from another entry file

[Wildcards are a bit tricky](https://github.com/Dschungelabenteuer/vite-plugin-entry-shaking/blob/main/RESOURCES.md#wildcardexports) but they're supposed to be supported and I think this could be easily fixed by recursively calling `resolveImportedEntities` whenever this scenario is encountered.  This scenario is already clearly identified by the `importDefault && name === '*'` condition within the `resolveImportedCircularEntities` function.  Your issue could be fixed a couple different ways but resolving final imports at _runtime_ instead of _analaysis-time_  looks better because it prevents overburdening the analysis time with entities that - in the end - may even not be used. So I'd rather go for a sort of _JIT_ approach. I'll setup a hotfix for this, but I feel like there might still be some uncovered edge case because, once again, [wildcards are a bit tricky](https://github.com/Dschungelabenteuer/vite-plugin-entry-shaking/issues/34#issuecomment-1815494589)

**Comment by Dschungelabenteuer** [comment]
I've released [0.5.0](https://github.com/Dschungelabenteuer/vite-plugin-entry-shaking/releases/tag/vite-plugin-entry-shaking%400.5.0) that should address this issue. Feel free to re-open if needed :)

**Comment by hwallis93** [comment]
Amazing thanks, I'll try it out tomorrow
